include/bbp/sonata/reports.h

```diff
@@ -68,15 +68,34 @@ void sonata_setup_communicators();
  */
 void sonata_set_min_steps_to_record(int steps);
 
+/**
+ * \brief Create a spike file
+ */
+void sonata_create_spikefile(const char* output_dir);
+
 /**
  * \brief Spike arrays to be written to file
  */
-void sonata_write_spikes(const char* population_name,
-                         const double* spike_timestamps,
-                         uint64_t num_timestamps,
-                         const int* spike_node_ids,
-                         uint64_t num_node_ids,
-                         const char* output_dir);
```

Comment:
Do we need to remove this pubic API?

Reply:
I think this API will not be used any more except for backward compatibility with neurodamus-core and coreneuron? But we will make a new release of all dependencies for the right API.

Reply:
Without much context, seems to me we should keep it and mark it as deprecated: `[[deprecated]]` 
https://en.cppreference.com/w/cpp/language/attributes/deprecated

Reply:
In general I'm not in favor of keeping deprecated code (especially since we ship the code with spack and can chose the version of libsonatareport to use depending on the neurodamus version). 

I guess in this case we can keep the function with the [[deprecated]] since is not a big function to implement with the new methods, but I will try to avoid this in the future with more complex code.

Reply:
Ok, I've added `[[deprecated]]`.

Reply:
Just saw an issue with installing neurodamus-core with this branch
```
In file included from SonataReportHelper.c(296):
/gpfs/bbp.cscs.ch/home/weji/workdir/blueconfigs/BUILD_HOME/software/linux-rhel7-x86_64/intel-19.1.2.254/libsonata-report-develop-kyfzhr/include/bbp/sonata/reports.h(79): error: expected a declaration
  [[deprecated("Use sonata_write_spike_populations() and relevant functions instead ")]] void
```
Forgot that there is a c file calling this header.  So how shall I use a c preferred declaration ?  @alexsavulescu @ferdonline  @jorblancoa

Reply:
So it's not actually deprecated? :) 
Well I can think of three approaches:

- just spack it
As @jorblancoa said, we can manage the deprecation via spack. Non-spack clients (are there any?) will complain the function is not there anymore. Just fine if we bump the version accordingly.

- playing nice:

If we want to let users know it's deprecated and we want them to use the new API then we might have to do something like (I assume we can have C++ and C users?): 
```cpp
#if defined(__cplusplus)
 [[deprecated]]
 void sonata_write_spikes(...);
#else
 void sonata_write_spikes(...) __attribute__((deprecated));
#endif
```
If we only have C users then just: 
```cpp
void sonata_write_spikes(...) __attribute__((deprecated));
```
However the `__attribute__((deprecated))` may not be supported by some compilers. This can get problematic, i.e. need to do something like https://stackoverflow.com/questions/295120/c-mark-as-deprecated/21265197#21265197 

- C API
This function will remain as-is, warning-free, as part of the C API and will call the new API behind the scenes. Double maintenance.

My money is on the first approach.

Reply:
`[[deprecated]]` is recent standard. My point is, we don't need to necessarily mark it deprecated, I don't think the functionality is that big that is gonna mess the source file. IMO there's old API, i.e. API that some users may want to use, and there's API that is deprecated because it should be removed (because it's unsafe, hard to maintain, etc). I think our case belongs to the first.
On the long run, sure, we can delete it to keep the lib fit, but I would at least give some time for depending projects to move and then major bump the version.

Reply:
I see, then let's keep it as-is, part of our C API.